### PR TITLE
[CT] Fix updateUser method

### LINKED ISSUE #4863: updating user in the composable breaks the user value
## Current behavior
When updateUser is called its running factoryparams method and assigns the returned value to the user.value.
https://github.com/DivanteLtd/vue-storefront/blame/next/packages/commercetools/composables/src/useUser/factoryParams.ts#L32

But, instead of the updated user object it returns an object with these keys: 
```
{currentUser: ..., updatedUserData: ...}
```

Here: https://github.com/DivanteLtd/vue-storefront/blob/next/packages/core/core/src/factories/useUserFactory.ts#L35

## Expected behavior

It will do the update properly.

## Steps to reproduce the issue

See description.

## Version of Vue Storefront

- [ ] Vue Storefront
- [x] Vue Storefront Next

## Can you handle fixing this bug by yourself?

- [ ] YES
- [x] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer. (doesn't apply to Next)
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ ] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details

- Browser: 
- OS: 
- Node: 
- Code Version:

### PR DESCRIPTION
**Pull request by filipsobol**
Closes #4863

### CODE REVIEW

**Review by andrzejewsky — APPROVED** [review]
@filipsobol update changelog

File: packages/commercetools/api-client/src/helpers/customer/index.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const changeCustomerEmail = (email: string) => ({
```
Comment (filrak) [inline packages/commercetools/api-client/src/helpers/customer/index.ts:1]: for what we need those? Is this a convention in the api client? I don't think these functions bring any value by being externalized, they're one-liners

Comment (andrzejewsky) [reply packages/commercetools/api-client/src/helpers/customer/index.ts:1]: @filrak this is an internal helper, it's not exposed to the external world

Comment (andrzejewsky) [reply packages/commercetools/api-client/src/helpers/customer/index.ts:1]: @filrak they are action creator actually here is the other example: https://github.com/DivanteLtd/vue-storefront/blob/next/packages/commercetools/api-client/src/helpers/cart/actions.ts, @filipsobol please ad *Action ad the end.

Comment (filrak) [reply packages/commercetools/api-client/src/helpers/customer/index.ts:1]: i know its a private api but I'm askig about the reason behind externalziing a one line fn into a separate file?

Comment (filipsobol) [reply packages/commercetools/api-client/src/helpers/customer/index.ts]: IMO in this case it only overcomplicates things.

**Pull request merged**